### DISCUSSION TITLE
Suggestions on pso optimization

### DISCUSSION THREAD

**Discussion by mh-tanver** [AstroBridge/BDLensing#139] [Q&A] [answered]
Although I got a chi square value of 1.1272858299044604. Is this a good fit for pso optimization?

Here is the collab link, https://colab.research.google.com/drive/1sQNss-zPEOCacjZpU3dpEjsxRRBoyr8Q?authuser=1#scrollTo=16fb38e8

![download](https://github.com/ajshajib/BDLensing/assets/93901410/91cd0285-708b-4bc9-b3c2-61f69fd564a2)
![download (1)](https://github.com/ajshajib/BDLensing/assets/93901410/aa92e2fa-877a-41f2-bb8a-e1112b2db615)

**Comment by rafee-adnan** [comment, marked as answer]
Following suggestions may improve the model:
- You can extend the region of masking at 1'o clock for a better chi squared value. As you can see from the residual that there are blue areas leaking out from the mask. The same goes for the mask at 11'o clock.
- inside `model_plot.source_plot()` add these two `with_caustics=True, scale_size=0.5`. Adjust the scale size according to the size of your reconstructed source.
- Otherwise, the model itself and the residual plot looks good! (I tried to see the values of your pso fit in colab, but I think it is restricted)

**Reply by mh-tanver** [reply]
Thanks a lot. Yes, colab was restricted then, didn't notice. It's accessible now.
